discovery-provider/src/premium_content/premium_content_types.py

```diff
@@ -8,7 +8,9 @@
 # will be gated.
 # They should match the PremiumConditions property in the track schema
 # in src/schemas/track_schema.json
-PremiumContentConditions = Literal["nft_collection", "follow_user_id", "tip_user_id"]
+PremiumContentConditions = Literal[
+    "nft_collection", "follow_user_id", "tip_user_id", "usdc_purchase"
```

Comment:
nit: maybe use the `USDC_PURCHASE_KEY` constant here?

Reply:
types wouldn't let me :(